alea/models/blueice_extended_model.py

```diff
@@ -178,42 +178,29 @@ def get_expectation_values(self, per_likelihood_term=False, **kwargs) -> dict:
             dict: Dictionary of expectation values. If per_likelihood_term is True, the dictionary
                 has the form {likelihood_name: {source_name: expectation_value, ...}, ...}.
 
-        Caution:
-            The function silently drops parameters it can't handle!
-
         Todo:
-            Current implementation is not elegant.
-            It copied the llh and sets the data to the copied llh,
-            because the call of llh needs data to be set.
-            But data is not needed for the expectation values.
-            We should update this function in the future after we stop using blueice.
-
             Make a self.likelihood_temrs dict with the likelihood names as keys and
             the corresponding likelihood terms as values.
 
         """
-        generate_values = self.parameters(**kwargs)  # kwarg or nominal value
         ret = cast(Dict[str, Dict[str, float]], {})
-
-        # calling ll need data to be set
-        self_copy = deepcopy(self)
-        self_copy.data = self_copy.generate_data()
+        # prepare generate_values
+        if not self.parameters.values_in_fit_limits(**kwargs):
+            raise ValueError("Values are not within fit limits")
+        generate_values = self.parameters(**kwargs)
 
         # ancillary likelihood does not contribute
-        for ll_term, ll_name, parameter_names, livetime_parameter in zip(
-            self_copy.likelihood_list[:-1],
-            self_copy.likelihood_names[:-1],
-            self_copy.likelihood_parameters,
-            self_copy.livetime_parameter_names,
+        for ll_name, lt_name in zip(
+            self.likelihood_names[:-1],
+            self.livetime_parameter_names,
         ):
             ret[ll_name] = {}
-            # WARNING: This silently drops parameters it can't handle!
-            call_args = {k: i for k, i in generate_values.items() if k in parameter_names}
-            if livetime_parameter is not None:
-                call_args["livetime_days"] = generate_values[livetime_parameter]
-
-            mus = ll_term(full_output=True, **call_args)[1]
-            for n, mu in zip(ll_term.source_name_list, mus):
+            ll_index = self.likelihood_names.index(ll_name)
+            lt = generate_values.pop(lt_name, None)
+            # compute the mus
+            self.data_generators[ll_index].compute_pdfs_and_mus(**generate_values, livetime_days=lt)
```

Comment:
OK, I tried but the simulated mean number of events has not changed. I guess because you also run `compute_pdfs_and_mus` here: https://github.com/XENONnT/alea/pull/140/files#diff-39b06664176349cdd066c85c1d6acf86bf7b094715d1f8b115395c2cf8abb9f2R99, right?